### PR TITLE
Rework boolean switch for new UI

### PR DESCRIPTION
**Pull request by tblivet**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | The Boolean switch has been completely reworked for the new UI.
| Type?             | refactor
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | @PrestaShopCorp
| How to test?      | Check if the 'Render field' → 'Boolean' works properly in Storybook.
||image

### CODE REVIEW

**Review by ga-devfront — APPROVED** [review]
image

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
